tests/components/octoprint/test_button.py

```diff
@@ -192,3 +192,78 @@ async def test_stop_job(hass: HomeAssistant) -> None:
         )
 
         assert len(stop_command.mock_calls) == 0
+
+
+async def test_shutdown_system(hass: HomeAssistant) -> None:
+    """Test the shutdown system button."""
+    await init_integration(hass, BUTTON_DOMAIN)
+
+    coordinator: OctoprintDataUpdateCoordinator = hass.data[DOMAIN]["uuid"][
+        "coordinator"
+    ]
+
+    # Test shutting down the system
+    with patch("pyoctoprintapi.OctoprintClient.shutdown") as shutdown_command:
+        coordinator.data["printer"] = OctoprintPrinterInfo(
+            {"state": {"flags": {}}, "temperature": []}
+        )
+        await hass.services.async_call(
+            BUTTON_DOMAIN,
+            SERVICE_PRESS,
+            {
+                ATTR_ENTITY_ID: "button.octoprint_shutdown_system",
+            },
+            blocking=True,
+        )
+
+        assert len(shutdown_command.mock_calls) == 1
```

Comment:
You should also freeze time and then check that the buttons state is equal to that time. 

Buttons states are equal to when they were last pressed, so by checking that, you can be sure that it successfully ran.